powermon/commands/reading_definition.py

```diff
@@ -121,96 +120,134 @@ def from_config(cls, reading_definition_config: dict, i) -> "ReadingDefinition":
+                return ReadingDefinitionDefault(
                     index=index,
+                    name=name,
+                    response_type=response_type,
                     description=description,
-                    options=options,
                     device_class=device_class,
                     state_class=state_class,
                     icon=icon,
+                    unit="A"
                 )
-
-            case ReadingType.ENFLAGS:
-                flags: dict[str, dict[str, str]] = reading_definition_config[3]
-                return ReadingDefinitionENFlags(
+            case ReadingType.PERCENTAGE:
+                return ReadingDefinitionDefault(
                     index=index,
+                    name=name,
+                    response_type=response_type,
                     description=description,
-                    enflags=flags,
                     device_class=device_class,
                     state_class=state_class,
                     icon=icon,
+                    unit="%"
                 )
-
-            case ReadingType.FLAGS:
-                flags: list[str] = reading_definition_config[3]
-                return ReadingDefinitionFlags(
+            case ReadingType.FREQUENCY:
+                return ReadingDefinitionDefault(
                     index=index,
+                    name=name,
+                    response_type=response_type,
                     description=description,
-                    flags=flags,
                     device_class=device_class,
                     state_class=state_class,
                     icon=icon,
+                    unit="Hz"
                 )
-
-            case ResponseType.INFO:
-                template = reading_type.split(":", 1)[1]
-                return ReadingDefinitionInfo(
+            case ReadingType.WATTS:
+                return ReadingDefinitionDefault(
                     index=index,
+                    name=name,
+                    response_type=response_type,
                     description=description,
-                    template=template,
                     device_class=device_class,
                     state_class=state_class,
                     icon=icon,
+                    unit="W"
                 )
 
             case _:
                 raise ValueError(
                     f"Reading description: {description} has unknown reading_type definition type: {reading_type}"
                 )
 
+class ReadingDefinitionDefault(ReadingDefinition):
+    def __init__(
+        self,
+        index: int,
+        name: str,
+        response_type: str,
+        description: str,
+        device_class: str = None,
+        state_class: str = None,
+        icon: str = None,
+        unit: str = ""
+    ):
+        super().__init__(index, name, response_type, description, device_class, state_class, icon, unit=unit)
+
+        
+    def is_valid_response(self, value) -> bool:
```

Comment:
do we need this as a separate function? can it be part of `translate_raw` possibly passing ResponseType so the processing can do more work or raise exception for invalid

Reply:
I was thinking it would contain validation checks, like temperature has to be a float and can't be negative if it's in Kelvin. Having it separate also helps with testing eventually. Currently it's under utilized and I'll remove them to make the code cleaner.